app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java

```diff
@@ -35,31 +37,58 @@
     public String uuid;
 
     public static TransmitterData create(byte[] buffer, int len, Long timestamp) {
-                StringBuilder data_string = new StringBuilder();
-        if (len < 6) { return null; };
-
-        for (int i = 0; i < len; ++i) {
-            data_string.append((char) buffer[i]);
-        }
-        String[] data = data_string.toString().split("\\s+");
-
-        randomDelay(100, 2000);
-        TransmitterData lastTransmitterData = TransmitterData.last();
-        if (lastTransmitterData != null && lastTransmitterData.raw_data == Integer.parseInt(data[0]) && Math.abs(lastTransmitterData.timestamp - timestamp) < (10000)) { //Stop allowing duplicate data, its bad!
+        if (len < 6) {
```

Comment:
Probably has. I have not gone deeply through the code since I put the dexbridge stuff in. Will review. 

 
John Stevens
M - +61 418526349

> On 20 May 2015, at 06:44, tzachi-dar <notifications@github.com> wrote:
> 
> In app/src/main/java/com/eveningoutpost/dexdrip/Models/TransmitterData.java:
> 
> > @@ -35,31 +37,58 @@
> >      public String uuid;
> >  
> >      public static TransmitterData create(byte[] buffer, int len, Long timestamp) {
> > -                StringBuilder data_string = new StringBuilder();
> > -        if (len < 6) { return null; };
> > -
> > -        for (int i = 0; i < len; ++i) {
> > -            data_string.append((char) buffer[i]);
> > -        }
> > -        String[] data = data_string.toString().split("\\s+");
> > -
> > -        randomDelay(100, 2000);
> > -        TransmitterData lastTransmitterData = TransmitterData.last();
> > -        if (lastTransmitterData != null && lastTransmitterData.raw_data == Integer.parseInt(data[0]) && Math.abs(lastTransmitterData.timestamp - timestamp) < (10000)) { //Stop allowing duplicate data, its bad!
> > +        if (len < 6) {
> seems like if(len<6) was already checked in this function.
> 
> —
> Reply to this email directly or view it on GitHub.
>